src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingQueryMessageWireProtocol.cs

```diff
@@ -300,6 +300,15 @@ private TCommandResult ProcessReply(ConnectionId connectionId, ReplyMessage<RawB
                     throw new MongoCommandException(connectionId, message, _command, materializedDocument);
                 }
 
+                if (rawDocument.Contains("writeConcernError"))
```

Comment:
Is there a helper that we could put this in? I think it's the same as the command path.

Reply:
Currently the two classes that handle commands (using either OP_MSG or OP_QUERY) don't really share much code.

How much effort do you want to see put into this? (Considering also that it's a point release).

Seems like the natural way to share code here is to create a base class that the two command protocol classes would derive from and put the common code in the base class.